e2e/testcases/otel_collector_test.go

```diff
@@ -0,0 +1,150 @@
+			fmt.Sprintf("iam.gke.io/gcp-service-account=%s@%s.iam.gserviceaccount.com", MonitorGSA, *e2e.GCPProject))
+	}
+
+	nt.T.Log("Restart otel-collector pod to refresh the ConfigMap, log and IAM")
+	nomostest.DeletePodByLabel(nt, "app", "opentelemetry", false)
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := validateOtelCollectorStatusCurrent(nt, false); err != nil {
+		nt.T.Fatal()
+	}
+
+	nt.T.Log("Wait 1 minute for metric exporter to settle")
+	time.Sleep(1 * time.Minute)
+	nt.T.Log("Checking the otel-collector log contains no failure...")
+	err := CheckDeploymentLogHasNoFailure(nt, ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Apply custom otel-collector ConfigMap that could cause duplicate time series error")
+	nt.MustKubectl("apply", "-f", "../testdata/otel-collector/otel-cm-duplicate-timeseries.yaml")
+	nt.T.Log("Restart otel-collector pod to refresh the ConfigMap and log")
+	nomostest.DeletePodByLabel(nt, "app", "opentelemetry", false)
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := validateOtelCollectorStatusCurrent(nt, true); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("wait 1 minute for metric exporter to settle")
+	time.Sleep(1 * time.Minute)
+	nt.T.Log("Checking the otel-collector log contains failure...")
+	if err := CheckDeploymentLogHasFailure(nt, ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace); err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func CheckDeploymentLogHasFailure(nt *nomostest.NT, deployment, namespace string) error {
+	nt.T.Helper()
+
+	args := []string{"logs", fmt.Sprintf("deployment/%s", deployment), "-n", namespace}
+	cmd := fmt.Sprintf("kubectl %s", strings.Join(args, " "))
+	out, err := nt.Shell.Kubectl(args...)
+	if err != nil {
+		nt.T.Logf("failed to run %q: %v\n%s", cmd, err, out)
+		return err
+	}
+
+	entry := strings.Split(string(out), "\n")
+	for _, m := range entry {
+		if strings.Contains(m, "failed to export time series to GCM") {
+			return nil
+		}
+	}
+	return fmt.Errorf("error expected in the log of deployment %s, namespace %s but found none", deployment, namespace)
+}
+
+func CheckDeploymentLogHasNoFailure(nt *nomostest.NT, deployment, namespace string) error {
+	nt.T.Helper()
+
+	args := []string{"logs", fmt.Sprintf("deployment/%s", deployment), "-n", namespace}
+	cmd := fmt.Sprintf("kubectl %s", strings.Join(args, " "))
+	out, err := nt.Shell.Kubectl(args...)
+	if err != nil {
+		nt.T.Logf("failed to run %q: %v\n%s", cmd, err, out)
+		return err
+	}
+
+	entry := strings.Split(string(out), "\n")
+	for _, m := range entry {
+		if strings.Contains(m, "failed to export time series to GCM") {
+			return fmt.Errorf("failure found in the log of deployment %s, namespace %s: %s", deployment, namespace, m)
+		}
+	}
+	return nil
+}
+
+// func validateDeploymentstatuscurrent
+func validateOtelCollectorStatusCurrent(nt *nomostest.NT, customConfigMap bool) error {
```

Comment:
The `Validate` functions in the test scaffolding typically don't wait for the condition to be satisfied - rather they just check that the condition is currently satisfied.

Suggest renaming this to something like `waitForOtelCollectorStatusCurrent` or `watchForOtelCollectorStatusCurrent`

Reply:
Name changed to `watchFor...`